gluish/common.py

```diff
@@ -64,3 +130,215 @@ def run(self):
+    """
+    date = ClosestDateParameter(default=datetime.date.today())
+    solruri = luigi.Parameter()
+    solrcore = luigi.Parameter()
+    sourceid = luigi.Parameter()
+    taskdir = luigi.Parameter()
+    salt = luigi.Parameter()
+    uritemplate = '{solruri}{solrcore}/update?commit=true'
+    headers = {'Content-Type': 'text/xml'}
+    payloadtemplate = '<delete><query>source_id:{sourceid}</query></delete>'
+
+    def requires(self):
+        return Available(service=self.solruri,
+                         message="provide a running Solr, please")
+
+    def run(self):
+        uri = self.uritemplate.format(solruri=self.solruri,
+                                      solrcore=self.solrcore)
+        payload = self.payloadtemplate.format(sourceid=self.sourceid)
+        result = {}
+
+        try:
+            response = requests.post(uri, data=payload, headers=self.headers)
+
+            result['status_code'] = response.status_code
+            result['content'] = response.content.decode('utf-8')
+
+            if result['status_code'] != 200:
+                raise SolrCleanupError('could not successfully clean Solr core \'%s\' (status code = \'%d\')' % (
+                    uri, response.status_code))
+
+            if not str(result['content']).startswith(
+                    '<?xml version=\"1.0\" encoding=\"UTF-8\"?>\n<response>\n<lst name=\"responseHeader\"><int name=\"status\">0</int>'):
+                raise SolrCleanupError(
+                    'could not successfully clean Solr core \'%s\' (response = \'%s\')' % (uri, result['content']))
+        except requests.exceptions.ConnectionError as error:
+            result['status_code'] = 404
+            result['error'] = str(error).replace('"', '\\"')
+
+            raise HTTPError('could not successfully establish connection to Solr index \'%s\'' % uri)
+        finally:
+            with self.output().open('w') as outfile:
+                outfile.write(json.dumps(result, indent=None) + "\n")
+
+    def output(self):
+        filename = 'response_' + str(self.salt) + '.json'
+        return luigi.LocalTarget(path=os.path.join(self.taskdir, filename))
+
+    def complete(self):
+        # TODO maybe define a better complete criteria (?) (background: complete() will be executed before run())
+        # e.g. check, whether something of this source is in the index
+        # or move all the stuff from run to complete (?)
+        if not self.output().exists():
+            return False
+
+        with self.output().open('r') as resultfile:
+            result = json.loads(resultfile.read())
+
+            if result['status_code'] != 200:
+                return False
+
+            if not str(result['content']).startswith(
+                    '<?xml version=\"1.0\" encoding=\"UTF-8\"?>\n<response>\n<lst name=\"responseHeader\"><int name=\"status\">0</int>'):
+                return False
+
+            return True
+
+
+class FillSolrIndex(luigi.Task):
```

Comment:
Yes, this is a task that is obviously missing in siskin, because we allowed ourselves to do this last step manually.

Reply:
I wonder, whether this functionality would fit better as a flag in solrbulk itself (similar to esbulk, where there is a [purge](https://github.com/miku/esbulk#usage) flag). This task basically says: Index data from (input) file and remove some data from SOLR before indexing. There is no data dependency, since the input is manually given. The purge-before-you-index or even purge-by-query-before-you-index functionality could be useful on it's own.